src/ServiceProvider.php

```diff
@@ -55,14 +57,28 @@ protected function registerAgent(): void
         });
     }
 
-    protected function listenForBooted(): void
+    protected function listenForEvents(): void
     {
+        $start_time = $this->app['request']->server('REQUEST_TIME_FLOAT') ?? microtime(true);
         $timeline_collector = $this->app->make(Agent::class)->getCollector(TimelineDataCollector::getName());
+        
+        // Application and Laravel startup times
+        $timeline_collector->startMeasure('app_boot', 'app', 'boot', 'App boot', $start_time);
+        $this->app->booting(function () use ($timeline_collector) {
+            $timeline_collector->stopMeasure('app_boot');
+            $timeline_collector->startMeasure('laravel_boot', 'laravel', 'boot', 'Laravel boot');
```

Comment:
What's the difference between `app_boot` and `laravel_boot`?

Reply:
We might need a better naming indeed. If I understand correctly how Laravel works, the difference between both of them is as follows:
- `app_boot`:  the time for loading Composer autoloader file, create Laravel application instance, create the container and register service providers.
- `laravel_boot`: load environment variables, register facades, resolve the Kernel, register routes, register middlewares...